src/vconsole/vconsole-setup.c

```diff
@@ -328,48 +320,88 @@ static void setup_remaining_vcs(int fd, bool utf8) {
         }
 }
 
+static int find_source_vc(char **ret_path, int *ret_idx, const char *src_vc) {
+        int ret_fd;
+
+        if (src_vc) {
+                _cleanup_close_ int fd = -1;
+
+                if (!(*ret_path = malloc(strlen(src_vc) + 1))) {
+                        log_oom();
+                        return -1;
+                }
+                fd = open_terminal(src_vc, O_RDWR|O_CLOEXEC);
+                if (fd < 0) {
+                        log_error_errno(fd, "Failed to open %s: %m", src_vc);
+                        return -errno;
```

Comment:
open_terminal() returns the error as return value, and you shouldn't use errno here. Also log_error_errno() returns the error passed to it as negative value, hence you can shorten this:

```c
if (fd < 0)
        return log_error_errno(fd, "....");
```

and the right thing will happen